ui/app/serializers/sync/destination.js

```diff
@@ -55,10 +60,11 @@ export default class SyncDestinationSerializer extends ApplicationSerializer {
     } else if (payload?.data) {
       // uses name for id and spreads connection_details object into data
       const { data } = payload;
-      const connection_details = payload.data.connection_details || {};
+      const { connection_details, options } = data || {};
       data.id = data.name;
       delete data.connection_details;
-      return { data: { ...data, ...connection_details } };
+      delete data.options;
+      return { data: { ...data, ...connection_details, ...options } };
```

Comment:
Not sure if connection_details will ever by falsy, but this unlike the previous version does not default the value to an empty object, so the spread may fail if it's null.

Reply:
Spreads just ignore null values

Reply:
To Chelsea's point the same could be true for `options`.